src/SourceCode.Clay.AspNetCore.Middleware.Correlation/CorrelationOptions.cs

```diff
@@ -13,29 +14,24 @@ public class CorrelationOptions
         /// <summary>
         /// The default http header to use.
         /// </summary>
-        internal const string DefaultHeader = "X-Correlation-ID";
+        public const string DefaultHeader = "X-Correlation-ID";
 
         /// <summary>
-        /// Gets or sets whether the <see cref="HttpContext.TraceIdentifier"/> of the context should be used if not correlation id
+        /// Gets or sets whether the <see cref="HttpContext.TraceIdentifier"/> of the context should be used if no correlation ID
         /// is present in request (defaults to false).
         /// If <c>false</c>, <see cref="CorrelationIdGenerator"/> is invoked to generate the correlation id.
         /// </summary>
-        public bool UseTraceIdentifier { get; internal set; } = false;
+        public bool UseTraceIdentifier { get; set; } = false;

```

Comment:
This should move to the CorrelationHeader struct so each header can be handled on it's own